lib/util/git/index.ts

```diff
@@ -1122,6 +1123,42 @@ export async function fetchBranch(
   }
 }
 
+/**
+ *
+ * Configure git to be able to use the git-credential feature
+ *
+ * 0. Run the command: git config credential.helper store
+ * 1. Creates, if missing, the file localDir/.git-credentials and fill it with `https://oauth2:<TOKEN>@<ENDPOINT>
+ *
+ */
+export async function configureCredentialHelperStore(
+  endpoint: string,
+  token: string
+): Promise<void> {
+  logger.debug('Configuring credential helper and git-credential file');
+  try {
+    await git.raw(['config', 'credential.helper', 'store']);
+    const localDir = GlobalConfig.get('localDir')!;
+    const gitCredentialsFile = upath.join(localDir, '.git-credentials');
+    if (await fs.pathExists(gitCredentialsFile)) {
+      logger.debug(
+        { gitCredentialsFile },
+        'The git-credentials file already exists'
+      );
```

Comment:
Throw error here if it can't be done

Reply:
There is a try/catch around that, lint complains about `throw of exception caught locally`

Reply:
Then narrow the existing try/catch, or rethrow it. You must throw a graceful error if this fails

Reply:
@rarkins like this ? 

```js
export async function configureCredentialHelperStore(
    endpoint: string,
    token: string
): Promise < void > {
    logger.debug('Configuring credential helper and git-credential file');

    try {
        await git.raw(['config', 'credential.helper', 'store']);
    } catch (err) {
        logger.debug({err}, 'Failed to configure git credentials.helper');
        throw new Error(PLATFORM_GIT_CREDENTIALS_ERROR);
    }

    const localDir = GlobalConfig.get('localDir') !;
    const gitCredentialsFile = upath.join(localDir, '.git-credentials');

    if (await fs.pathExists(gitCredentialsFile)) {
        logger.debug({gitCredentialsFile},'The .git-credentials file already exists');
        throw new Error(PLATFORM_GIT_CREDENTIALS_ERROR);
    } else {
        const contents: string = `https://oauth2:${token}@${endpoint}`;
        logger.debug({gitCredentialsFile,contents},'Writing credentials to file');

        try {
            await fs.outputFile(gitCredentialsFile, contents, {mode: 0o640});
        } catch (err) {
            logger.debug({err}, 'Failed to write .git-credentials file');
            throw new Error(PLATFORM_GIT_CREDENTIALS_ERROR);
        }
    }
}
```